src/state/orderPlacement/reducer.ts

```diff
@@ -44,6 +48,13 @@ export default createReducer<SwapState>(initialState, (builder) =>
         sellAmount,
       }
     })
+    .addCase(invertPrice, (state) => {
+      return {
+        ...state,
+        price: getInverse(Number(state.price), 16).toString(),
+        showPriceInverted: !state.showPriceInverted,
+      }
```

Comment:
This is the correct place to have this kind of logic but seems repeated a few times across this pull request. `useSwapPrice` can be modified, or a selector for price can be used to avoid that.  Something to take into account at the moment of a refactor.

Reply:
It's tricky. Probably, the most tricky part of this PR.

I am actually not sure what the best way is to store the price. Maybe it would have been better to store it always in the canonical "form", - as we will submit it to the contract. 
Here I am storing it, as the user inserts the value, but this forces me to later change the values again for contract submission.

Let me think about it a little bit more and then probably, I will refactor it